app/tests/api/logging/test_logging.py

```diff
@@ -0,0 +1,153 @@
+import json
+import logging
+import logging.config
+import re
+from contextlib import contextmanager
+
+import pytest
+from _pytest.logging import LogCaptureHandler
+from flask import g
+
+import api.logging
+from api.logging.log_formatters import JsonFormatter
+
+
+@pytest.fixture(autouse=True)
+def reset_logging_for_pytest():
+    """Reset the logging library so that later tests using caplog work correctly.
+
+    Our logging initialization code api.util.logging.init(), which is used in many tests
+    in this file, conflicts with pytest's caplog fixture.
```

Comment:
Fascinating. Why?

Reply:
I never quite figured out the full "why", but I think some of the configurations we do (which are effectively globally set) make the formatting and processing a bit "different" and parsing it in caplog ends up being more complex / not working (see the other class I built here to deal with propagation issues).